src/components/Table/types.ts

```diff
@@ -11,30 +11,45 @@ export type InternalExpansionIconProps = {
   onClick: MouseEventHandler;
 };
 
-export interface columnTypes {
-  [index: string]: {
-    name?: string;
-    width?: string;
-    minTableWidth?: number;
-    sortable?: boolean;
-    footerContent?: ReactNode;
-    sortFunction?: (item1: any, item2: any) => boolean;
-    isGroupLabel?: boolean;
-    cellComponent?: StyledSubcomponentType;
-    rowComponent?: StyledSubcomponentType;
-    headerCellComponent?: StyledSubcomponentType;
-    groupCellComponent?: StyledSubcomponentType;
-  };
+/** Map between column key strings and column properties. */
+export type Columns = Record<string, Column>;
+
+/** The properties that can be defined for a single column. */
+export interface Column {
+  name?: string;
+  width?: string;
+  minTableWidth?: number;
+  sortable?: boolean;
+  footerContent?: ReactNode;
+  sortFunction?: (item1: any, item2: any) => boolean;
+  isGroupLabel?: boolean;
+  cellComponent?: ComponentBuilder;
+  rowComponent?: ComponentBuilder;
+  headerCellComponent?: ComponentBuilder;
+  groupCellComponent?: ComponentBuilder;
 }
 
+type ComponentBuilder = StyledSubcomponentType | ((props: any) => JSX.Element) | React.FC;
```

Comment:
Didn't we move this to a separately exported type?

Reply:
Re-added to StyledSubcomponentType